src/components/Radios.tsx

```diff
@@ -0,0 +1,282 @@
+            checked={item.checked ? true : undefined}
+            disabled={item.disabled ? true : undefined}
+            data-aria-controls={
+              item.childrenWhenChecked ? `conditional-${id}` : undefined
+            }
+            aria-describedby={hasHint ? itemHintId : undefined}
+            {...radioExtraAttributes}
+            onChange={
+              onChange !== undefined
+                ? (): void => {
+                    //eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    onChange(item.value!);
+                  }
+                : undefined
+            }
+          />
+          <Label className={labelClasses} labelFor={id} {...item.label}>
+            {item.label && item.label.children}
+          </Label>
+          {hasHint && (
+            <Hint
+              id={itemHintId}
+              className={classNames(
+                "govuk-radios__hint",
+                item.hint && item.hint.className
+              )}
+            >
+              {item.hint && item.hint.children}
+            </Hint>
+          )}
+        </div>
+        {item.childrenWhenChecked && (
+          <div className={conditionalClasses} id={`conditional-${id}`}>
+            {item.childrenWhenChecked}
+          </div>
+        )}
+      </React.Fragment>
+    );
+  };
+
+  const renderItem = (item: RadioButton, index: number): JSX.Element => {
+    const radioExtraAttributes = Attributes.ariaAndData(item);
+    if (item.divider) {
+      return renderDivider(item, index, radioExtraAttributes);
+    } else {
+      return renderRadio(item, index, radioExtraAttributes);
+    }
+  };
+
+  const innerContent = (
+    <>
+      {radiosHint && hintComponent}
+      {errorMessage && errorMessageComponent}
+      <div
+        className={radioClassNames}
+        data-module={hasConditional ? "govuk-radios" : undefined}
+        {...extraAttributes}
+      >
+        {items.map((item, index) => renderItem(item, index))}
+      </div>
+    </>
+  );
+
+  return (
+    <FormGroup {...formGroup}>
+      {fieldset ? (
+        <Fieldset
+          describedBy={
+            describedBy.length > 0 ? describedBy.join(" ") : undefined
+          }
+          {...fieldset}
+        >
+          {innerContent}
+        </Fieldset>
+      ) : (
+        <>{innerContent}</>
+      )}
+    </FormGroup>
+  );
+};
+
+Radios.propTypes = {
+  className: PropTypes.string,
+  name: PropTypes.string.isRequired,
+  items: PropTypes.arrayOf(
+    PropTypes.exact({
+      id: PropTypes.string,
+      value: PropTypes.string,
+      label: PropTypes.object,
+      hint: PropTypes.object,
+      divider: PropTypes.string,
+      checked: PropTypes.bool,
+      childrenWhenChecked: PropTypes.node,
+      disabled: PropTypes.bool
+    })
+  ).isRequired,
+  fieldset: PropTypes.object,
+  hint: PropTypes.object,
+  errorMessage: PropTypes.object,
+  formGroup: PropTypes.object,
+  idPrefix: PropTypes.string
+};
```

Comment:
Let's be more specific with prop types than `PropTypes.object`. Suggest using `PropTypes.shape`, or even inheriting the prop types from other components. For example:

```
label: PropTypes.shape(Label.propTypes)
```

Reply:
Can we specify the types of the other things that are `PropTypes.object` too?